### ISSUE TITLE
Sphinx doesn't check an __init__ docstring signature

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#866]
For any C (or Cython) function you can do this:

```
def f(arg, optional_arg = None):
    """f(arg[, optional_arg])

    Blablabla."""

    pass
```

And autodoc will parse out the correct signature just fine.

However, if you try to do it for a class's **init** it won't work:

```
class A:
    def __init__(self, arg, optional_arg):
        """__init__(self, arg[, optional_arg])

        Blablabla."""

        pass
```

I think this is a bug and should be fixed. Looking into the code I see that FunctionDocumenter inherits from DocstringSignatureMixin which gives it it's signature-detection capabilities. However no such feature has been added for ClassDocumenter.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/866
- Originally reported by: Orson Peters
- Originally created at: 2012-01-29T19:15:29.137

**Comment by shimizukawa** [comment]
_From Orson Peters on 2012-01-29 19:00:42+00:00_

I have added a slightly hacky but working fix.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 08:30:16+00:00_

Thanks for the report, this was fixed by commit 4aa18c3dc8c6.
